lib/DDG/Spice/Holiday.pm

```diff
@@ -13,81 +13,80 @@ spice wrap_jsonp_callback => 0;
-    my $chosenHoliday;
+    my ($chosenYear, $chosenCountry, $chosenHoliday);
+
     my $userSpecifiedYear = 0;
 
+    # make sure we have a $loc and country name
+    return unless $loc and $loc->country_name;
+
+    # Current year and users location are the defaults unless otherwise specified by the query
+    $chosenYear = (localtime(time))[5] + 1900;
+    $chosenCountry = $loc->country_name;
+
     # Remove common prefixes queries for holidays may contain
-    $query =~ s/$prefix//;
+    $query =~ s/^(when|what day) is\s+//;
 
-    # Sanitize the query by replacing non-alphanumeric characters, eg: 
+    # Sanitize the query by replacing non-alphanumeric characters, eg:
     #   "st. patricks day" -> "st patricks day"
     #   "eid al-fitr" -> "eid al fitr"
     #   "father's day" -> "fathers day"
     $query =~ s/[^\w\s\-]//;
     $query =~ s/-/ /;
 
-    $query =~ s/$in(?<year>$year)//;
-    if ($+{year}) {
+    # Extract holiday from query
+    $query =~ s/(?<holiday>$triggers_re)//;
+    $chosenHoliday = $+{holiday};
+
+    # Extract year from query
+    # Do this after holiday because some holidays contain years
+    if ($query =~ s/$in ?(?<year>$year_re)//) {
         $chosenYear = $+{year};
+        # These are the min/max years available from the API (as of Feb 2016, API version 2)
+        return unless ($chosenYear >= 1600 && $chosenYear <= 2400);
         $userSpecifiedYear = 1;
-    } else {
-        $chosenYear = $defaultYear;
     }
 
-    $query =~ s/$inThe(?<country>$country)$//;
-    if ($+{country}) {
+    # Extract country from query
+    if ($query =~ s/$inThe ?(?<country>$country_re)$//) {
         $chosenCountry = $+{country};
-    } else {
-        $chosenCountry = $defaultCountry;
-    }
-     
-    # Load the list of holidays for the selected country
-    my $holidays;
-    my $holidayFile = "countries/" . country2code($chosenCountry) . ".txt";
-    if (-f "share/spice/holiday/" . $holidayFile) {    
-        $holidays = join('|', share($holidayFile)->slurp(chomp => 1));
-    } else {
-        return; # Unknown country
     }
 
-    $query =~ s/(?<holiday>$holidays)//;
-    if ($+{holiday}) {
-        $chosenHoliday = $+{holiday};
-    } else {
-        return; # Unknown holiday
-    }
+    # Verify we have a country code for chosen country
+    return unless my $countryCode = country2code($chosenCountry);
 
     # If there's anything left in the query we can't be sure its relevant
     return unless ($query =~ /^\s*$/);
-    
-    # These are the min/max years available from the API (as of Feb 2016, API version 2)
-    return unless ($chosenYear >= 1600 && $chosenYear <= 2400);
-    
-    return $chosenCountry, $chosenHoliday, $chosenYear, $userSpecifiedYear; 
+
+    # Load the list of holidays for the selected country
+    my $filePath = "countries/$countryCode.txt";
+
+    # Ensure we have a list of holidays for the chosen country
+    return unless (my @lines = share($filePath)->slurp(chomp => 1));
```

Comment:
Could we add a test to verify this works as intended? You'll need to specify a country that is listed in `countries.txt` but doesn't have a corresponding holiday file (because timeandate.com don't have data for it), eg: "when is christmas day in aland islands"